### PR TITLE
Order showcart api

### PR DESCRIPTION
**Pull request by SoniaChoo**

### CODE REVIEW

File: db/orders.go [hunk 1]
```diff
@@ -50,3 +50,80 @@ func UpdatePriceInOrderDetail(d *model.Dish) error {
...
+	db, err := DBFactory()
+	if err != nil {
+		log.Printf("error connect database, %v\n", err)
+		return nil, err
+	}
+
+	//start to excute SQL query
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	row, err := db.QueryContext(ctx, "select * from orders where user_id = ?", o.UserId)
```
Comment (ziyitony) [inline db/orders.go]: please specify the order status in SQL

File: http/orderHandler.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	orderids, err := db.GetOrderIdInTableOrder(&order)
+	if err != nil {
+		log.Printf("select by userid in order to get order_id with error, error is %s\n", err.Error())
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "select by userid = %d in table order failed")
+		return
+	}
+
+	//judge if len(details) = 1
+	if len(orderids) == 0 || len(orderids) > 1 {
```
Comment (ziyitony) [inline http/orderHandler.go]: please update the if condition

File: db/orders.go [hunk 3]
```diff
@@ -50,3 +50,80 @@ func UpdatePriceInOrderDetail(d *model.Dish) error {
...
+	db, err := DBFactory()
+	if err != nil {
+		log.Printf("error connect database, %v\n", err)
+		return nil, err
+	}
+
+	//start to excute SQL query
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	row, err := db.QueryContext(ctx, "select * from order_detail where order_id = ? and status = ?", order_id, InCartStatus)
```
Comment (ziyitony) [inline db/orders.go]: please update the SQL to make this function more reuseable.

File: db/orders.go [hunk 4]
```diff
@@ -50,3 +50,80 @@ func UpdatePriceInOrderDetail(d *model.Dish) error {
...
+	db, err := DBFactory()
+	if err != nil {
+		log.Printf("error connect database, %v\n", err)
+		return err
+	}
+
+	//start to excute SQL query
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	_, err = db.QueryContext(ctx, "update orders set total_price = ? where order_id = ? and status = ?", total_price, order_id, InCartStatus)
```
Comment (ziyitony) [inline db/orders.go]: same as above

**Review by ziyitony — CHANGES_REQUESTED** [review]

File: db/orders.go (hunk 1 above)
Comment (SoniaChoo) [reply db/orders.go]: updated

File: http/orderHandler.go (hunk 2 above)
Comment (SoniaChoo) [reply http/orderHandler.go]: updated

File: db/orders.go (hunk 3 above)
Comment (SoniaChoo) [reply db/orders.go]: updatedupdated

File: db/orders.go (hunk 4 above)
Comment (SoniaChoo) [reply db/orders.go]: updated

**Review by ziyitony — APPROVED** [review]

**Pull request merged**